src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -229,3 +239,246 @@ def getPFN(self, nodes=None, lfns=None, destination=None, protocol='srmv2', cust
+           * meta:      Meta-data associated with the data identifier is represented using key/value
+                        pairs in a dictionary.
+           * rules:     Replication rules associated with the data identifier. A list of dictionaries,
+                        e.g., [{"copies": 2, "rse_expression": "TIERS1"}, ].
+           * lifetime:  DID's lifetime (in seconds).
+        :return: a boolean to represent whether it succeeded or not
+        """
+        response = False
+        try:
+            # add_container(scope, name, statuses=None, meta=None, rules=None, lifetime=None)
+            response = self.cli.add_container(scope, name, **kwargs)
+        except DataIdentifierAlreadyExists:
+            self.logger.debug("Container name already exists in Rucio: %s", name)
+            response = True
+        except Exception as ex:
+            self.logger.error("Exception creating container: %s. Error: %s", name, str(ex))
+        return response
+
+    def createBlock(self, name, scope='cms', attach=True, **kwargs):
+        """
+        _createBlock_
+
+        Create a CMS block (Rucio dataset) in a given scope. It also associates
+        the block to its container
+        :param name: string with the block name
+        :param scope: optional string with the scope name
+        :param attach: boolean whether to attack this block to its container or not
+        :param kwargs:  supported keyword arguments (from the Rucio CLI API documentation):
+           * statuses:  Dictionary with statuses, e.g. {"monotonic":True}.
+           * lifetime:  DID's lifetime (in seconds).
+           * files:     The content.
+           * rse:       The RSE name when registering replicas.
+           * meta:      Meta-data associated with the data identifier. Represented
+                        using key/value pairs in a dictionary.
+           * rules:     replication rules associated with the data identifier. A list of
+                        dictionaries, e.g., [{"copies": 2, "rse_expression": "TIERS1"}, ].
+        :return: a boolean with the outcome of the operation
+
+        NOTE: we will very likely define a rule saying: keep this block at the location it's
+              being produced
+        """
+        response = False
+        try:
+            # add_dataset(scope, name, statuses=None, meta=None, rules=None, lifetime=None, files=None, rse=None)
+            response = self.cli.add_dataset(scope, name, **kwargs)
+        except DataIdentifierAlreadyExists:
+            self.logger.debug("Block name already exists in Rucio: %s", name)
+            response = True
+        except Exception as ex:
+            self.logger.error("Exception creating block: %s. Error: %s", name, str(ex))
+
+        # then attach this block recently created to its container
+        if response and attach:
+            container = name.split('#')[0]
+            response = self.attachDIDs(kwargs.get('rse'), container, name, scope)
```

Comment:
You can get an error here if the block already existed and was already attached to the container (I think).

Reply:
Yeah, you catch a similar error in the next section.

Reply:
Yes, the error handling is done on the `attachDIDs` method